File: frontend/lib/domain/entities/device.dart
class DeviceEntity {
  final int? id;
  final double? minSoilSetting;
  final double? maxSoilSetting;
  final bool? isOn;
  final double? latitude;
  final double? longitude;
  final String? location;
  final int? progressPlan;
  final int? progressNow;
  final String? code;
  final String? plantName;
  final bool? isActiveSoil;
  final bool? isActivePump;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const DeviceEntity({
    this.id,
    this.minSoilSetting,
    this.maxSoilSetting,
    this.isOn,
    this.latitude,
    this.longitude,
    this.location,
    this.progressPlan,
    this.progressNow,
    this.code,
    this.plantName,
    this.isActiveSoil,
    this.isActivePump,
    this.createdAt,
    this.updatedAt,
  });
}

File: Backend/models/device.go
package models

import (
	"time"
)

type Device struct {
	ID               string  `gorm:"type:text;primaryKey"`
	MinSoilSetting   float64 `gorm:"default:0.0"`
	MaxSoilSetting   float64 `gorm:"default:50.0"`
	IsOn             bool    `gorm:"default:false"`
	Latitude         float64 `gorm:"type:numeric(9,6);default:0.000000"`
	Longitude        float64 `gorm:"type:numeric(9,6);default:0.000000"`
	Location         string
	ProgressPlan     int `gorm:"default:0"`
	ProgressNow      int `gorm:"default:0"`
	PlantName        *string
	Users            []User `gorm:"many2many:user_devices;joinForeignKey:DeviceID;joinReferences:UserID"`
	PumpLogs         []PumpLog
	Alarms           []Alarm
	SensorAggregates []SensorAggregate
	CreatedAt        time.Time `gorm:"default:now()"`
	UpdatedAt        time.Time `gorm:"default:now()"`
}
